nav2_core/include/nav2_core/global_planner.hpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2019, Samsung Research America
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef NAV2_CORE_GLOBAL_PLANNER_H_
+#define NAV2_CORE_GLOBAL_PLANNER_H_
+
+#include <string>
+#include "rclcpp/rclcpp.h"
+#include "nav2_costmap_2d/nav2_costmap_2d.hpp"
+#include "tf2_ros/Buffer.h"
+#include "nav2_msgs/msg/path.h"
+#include "geometry_msgs/msg/pose_stamped.h"
+
+namespace nav2_core
+{
+
+/**
+ * @class GlobalPlanner
+ * @brief Abstract interface for global planners to adhere to with pluginlib
+ */
+class GlobalPlanner
+{
+public:
+  /**
+   * @brief Virtual destructor
+   */
+  virtual ~GlobalPlanner() {}
+
+  /**
+   * @param  parent pointer to user's node
+   * @param  name The name of this planner
+   * @param  tf A pointer to a TF buffer
+   * @param  costmap_ros A pointer to the costmap
+   */
+  virtual void configure(const rclcpp::Node * parent,
+    const std::string & name, tf2_ros::Buffer * tf,
+    nav2_costmap_2d::Costmap2DROS * costmap_ros) = 0;
```

Comment:
Do you think this could be replaced with some abstract world representation? How are you thinking this works for a planner that uses some other world representation?

Reply:
Please provide me an example of that. There's alot of good heritage for this. 

As I said, these are changing interfaces, but lets not over-engineer for things that don't exist

https://github.com/locusrobotics/robot_navigation/tree/master/nav_core2/include/nav_core2
https://github.com/ros-planning/navigation/tree/melodic-devel/nav_core/include/nav_core

Reply:
No where else in the stack do we support other world model representations. Why would it be necessary here and no where else?

Reply:
I agree with @SteveMacenski. If at a future date we add another world representation, we can extend the interface to have a configure method which accepts it.

Reply:
Marking as resolved